### ISSUE TITLE
Special use case: blend4web default code structure

### ISSUE THREAD

**Issue by AttilaVM** [mooz/js2-mode#315]
Well I found an easy workaround for this one, however maybe this issue can be useful. 

I am currently writing a blend4web application, and its default project code structure is very uncanny:

The whole app source code resides inside the argument list of the b4w.register() method in a "nested" function like

```
b4w.register("project-name", function register(exports, require) {
    // The app's source code... can be thousands of lines.
});

b4w.require("busHighPoly").init();
```

Maybe stuff like this an abomination only for me, because I used python before I began to learn JavaScript, not a long time ago.

Anyway the code inside the argument list is not  recognizable for js2-mode's js2-imenu-extras-mode. It can be easily worked around with assigning the nested function from the argument list to variable, then pass that variable to b4w.register. like this:

```
var application = function register(exports, require) {
    // app code
};

b4w.register("busHighPoly", application);
b4w.require("busHighPoly").init();
```

**Comment by dgutov** [comment]
`js2-imenu-extras-mode` should put all unrecognized functions into the `?` namespace. Doesn't it?

**Comment by AttilaVM** [comment]
I can check it tomorrow.

**Comment by AttilaVM** [comment]
@dgutov You are right it cleverly does, except constructor functions. However it recognize their methods:

![before_fix](https://cloud.githubusercontent.com/assets/12675000/13393124/b597e388-dedf-11e5-98d7-aed8dd3164a1.png)

The VectorMath constructor is not recognized, but it's add method is in the list.

**Comment by dgutov** [comment]
OK.

If you want me to do something about it, I'll need a self-contained example I can try.

**Comment by AttilaVM** [comment]
I provide a self-contained example, however personally I won't  need js2 modified for this, because defining a function in an other function argumentum list is a bit obscure way of programming for me. So even when I wrote my first post it was no longer a personal problem, but it maybe be useful for others.

[example.zip](https://github.com/mooz/js2-mode/files/150979/example.zip)

**Comment by dgutov** [comment]
Thanks. Here's the code snippet for easier reference:

```js
// Recognized
var outerFunction = function outerFunction(func) {
    func();
};

outerFunction(function argumentFunction() {
    // NOT recognized
    function ExampleConstructor() {
        // Recognized
        this.method = function exampleMethod() {
            document.write("Hello world");
        };
    };
    var exampleObj = new ExampleConstructor();
    exampleObj.method();
});
```

Looking into it, a coherent fix would not be trivial. You seem to already have worked that out, but just to be sure, you can replace `function ExampleConstructor` with `var ExampleContructor = function`, and it will be indexed.

**Comment by AttilaVM** [comment]
That works too, but I assigned the _argumentFunction_ to a variable, then passed it to the _outerFunction_.

```
// Recognized
var outerFunction = function outerFunction(func) {
    func();
};

// Recognized
var argumentFunction = function argumentFunction() {
    // recognized
    function ExampleConstructor() {
        // Recognized
        this.method = function exampleMethod() {
            document.write("Hello world");
        };
    };
    var exampleObj = new ExampleConstructor();
    exampleObj.method();
};

outerFunction(argumentFunction);
```

Then it works like a charm.
Thank you all for this mode, it helps me a lot.
